test.js

```diff
@@ -134,6 +134,50 @@ test('execa.shellSync() includes stdout and stderr in errors for improved debugg
 	t.throws(() => execa.shellSync('node fixtures/error-message.js'), {message: STDERR_STDOUT_REGEXP, code: 1});
 });
 
+test.cb('execa() with .kill after it with SIGKILL should kill cleanly', t => {
+	const proc = execa('node', ['fixtures/sub-process-no-killable']);
+	t.is(typeof proc.pid, 'number');
+	setTimeout(() => {
+		proc.kill('SIGKILL');
+	}, 100);
+
+	setTimeout(() => {
+		t.false(isRunning(proc.pid));
+		t.end();
+	}, 2000);
```

Comment:
Instead of using both `setTimeout()` you could maybe just:
  - call `proc.kill('SIGKILL') synchronously
  - wait for `await proc` rejection in a try/catch block (and asserts it is rejected)
  - asserts that `error.signal` is `SIGKILL`

Reply:
bb5b280c806e049600d3de4db876a7d5d6a17888